### DISCUSSION TITLE
Capture console output

### DISCUSSION THREAD

**Discussion by unruledboy** [sebastienros/jint#1086] [Q&A]
I searched all the issues (including closed) but could not find any discussion on how to capture console output, especially the outputs of following 4 actions

- console.info
- console.log
- console.warn
- console.error

How do we hook up with the logging?

**Comment by lahma** [comment]
I guess you could register you own custom `Console` object instance which has the methods/functions you need?

**Reply by unruledboy** [reply]
Yes, looks like the following does not work:

```
                Action<object> logger = (log) => Console.WriteLine(log));
                var logger = new ConsoleLogger();
                var engine = new Engine()
                    .SetValue("console.log", logger)
                    .Execute("console.log('foo')");

```

Had to do like this:
```
public class ConsoleLogger
    {
        public void Log(object log)
        {

        }
    }
                var logger = new ConsoleLogger();
                var engine = new Engine()
                    .SetValue("console", logger)
                    .Execute(request.Script);

```

Surprisingly, the function name is not case sensitive.

**Reply by lahma** [reply]
By default Jint matches case-sensitively, except for the first character for interop to be more seamless. This can be configured via engine interop options.
